examples-gallery/beginner/plot_wheel_on_bumpy_road.py

```diff
@@ -0,0 +1,419 @@
+
+h = sm.symbols('h')
+num_nodes = 301
+t0, tf = 0, h*(num_nodes - 1)
+interval = h
+
+par_map = {}
+par_map[m_car] = 350.0
+par_map[m_wheel] = 5.0
+par_map[g] = 9.81
+par_map[l_0] = 1.0
+par_map[r1] = 0.1
+par_map[r2] = 0.1
+par_map[r3] = 0.39
+par_map[r4] = 0.29
+par_map[r5] = 0.1
+par_map[k1] = 250000.0
+
+# %%
+# Plot the road.
+r11, r22, r33, r44, r55 = [par_map[key] for key in [r1, r2, r3, r4, r5]]
+rough_surface_lam = sm.lambdify((x_car, r1, r2, r3, r4, r5),
+                                rough_surface(x_car), cse=True)
+XX = np.linspace(0, 10, 100)
+r11, r22, r33, r44, r55 = [par_map[key] for key in [r1, r2, r3, r4, r5]]
+fig, ax = plt.subplots(figsize=(7, 2), layout='tight')
+ax.plot(XX, rough_surface_lam(XX, r11, r22, r33, r44, r55))
+ax.set_xlabel('[m]')
+ax.set_ylabel('[m]')
+_ = ax.set_title('Road Profile')
+# %%
+#To be minimized:
+# :math:`\int (\dfrac{d}{dt}uz_{car})^2 dt + \text{weight} \cdot t_f`
+# ``weight`` is a scalar that can be used to adjust the importance of the
+# speed.
+weight = 1.e9
+
+def obj(free):
+    uz_dot = np.sum([free[i]**2 for i in range(8*num_nodes, 9*num_nodes)])
+    return (uz_dot)*free[-1] + weight*free[-1]
+
+def obj_grad(free):
+    grad = np.zeros_like(free)
+    grad[9*num_nodes:10*num_nodes] = 2*free[8*num_nodes:9*num_nodes]*free[-1]
+    grad[-1] = (
+            + np.sum([free[i]**2 for i in range(8*num_nodes, 9*num_nodes)])
+            + weight
+        )
+    return grad
+
+# %%
+# Add the instance constraints and bounds.
+instance_constraints = (
+    x_car.func(t0) - 0.0,
+    ux_car.func(t0) - 0.0,
+    accel_street.func(t0) - 0.0,
+    accel_body.func(t0) - 0.0,
+    steady_body.func(t0) - 1.0,
+    x_car.func(tf) - 10.0,
+    ux_car.func(tf) - 0.0,
+)
+
+bounds = {
+    h: (0.0, 1.0),
+    x_car: (0.0, 10.0),
+    z_wheel: (0.0, 2.0),
+    ux_car: (0.0, np.inf),
+    prevent_jump: (0.0, 0.1),
+    steady_body: (0.85, 1.0),
+    c: (0.0, 750),
+    k: (15000, 100000),
+    fx: (-50000, 50000),
+    l_GW: (0.0, 1.0),
+}
+
+# %%
+# Use an existing solution if available, else iterate to find one.
+fname =f'quarter_car_on_bumpy_road_{num_nodes}_nodes_solution.csv'
+
+prob = Problem(obj,
+                obj_grad,
+                eom,
+                state_symbols,
+                num_nodes,
+                interval,
+                known_parameter_map=par_map,
+                instance_constraints=instance_constraints,
+                bounds=bounds,
+                time_symbol=t,
+                backend='numpy',
+            )
+
+if os.path.exists(fname):
+    # use the existing solution
+         solution = np.loadtxt(fname)
+else:
+    # Iterate to find the solution. As the convergence is not easy, one has to
+    # start with a smooth road and then increase the roughness gradually.
+    # Before the bounds have to be tightended gardually.
+    # Here backend = 'cython' is used, as it is faster than 'numpy'. for solve.
+    prob.backend = 'cython'
```

Comment:
There is no need to change the backend from numpy and this wouldn't do anything here anyways. The backend only affects the initialization of the object.